garak/generators/websocket.py

```diff
@@ -0,0 +1,410 @@
+        """Extract text from response using JSON field extraction"""
+        if not self.response_json:
+            return response
+        
+        try:
+            response_data = json.loads(response)
+            
+            # Handle JSONPath-style field extraction
+            if self.response_json_field.startswith('$'):
+                # Simple JSONPath support for common cases
+                path = self.response_json_field[1:]  # Remove $
+                if '.' in path:
+                    # Navigate nested fields
+                    current = response_data
+                    for field in path.split('.'):
+                        if isinstance(current, dict) and field in current:
+                            current = current[field]
+                        else:
+                            return response  # Fallback to raw response
+                    return str(current)
+                else:
+                    # Single field
+                    return str(response_data.get(path, response))
+            else:
+                # Direct field access
+                return str(response_data.get(self.response_json_field, response))
+                
+        except (json.JSONDecodeError, KeyError, TypeError):
+            logger.warning(f"Failed to extract JSON field '{self.response_json_field}', returning raw response")
+            return response
+
+    async def _connect_websocket(self):
+        """Establish WebSocket connection with proper error handling"""
+        try:
+            # Prepare connection arguments
+            connect_args = {
+                'open_timeout': self.connection_timeout,
+                'close_timeout': self.connection_timeout,
+            }
+            
+            # Add headers if provided
+            if self.headers:
+                connect_args['additional_headers'] = self.headers
+            
+            # SSL verification
+            if self.secure and not self.verify_ssl:
+                import ssl
+                connect_args['ssl'] = ssl.create_default_context()
+                connect_args['ssl'].check_hostname = False
+                connect_args['ssl'].verify_mode = ssl.CERT_NONE
+            
+            logger.debug(f"Connecting to WebSocket: {self.uri}")
+            self.websocket = await websockets.connect(self.uri, **connect_args)
+            logger.info(f"WebSocket connected to {self.uri}")
+            
+        except Exception as e:
+            logger.error(f"Failed to connect to WebSocket {self.uri}: {e}")
+            raise
+
+    async def _send_and_receive(self, message: str) -> str:
+        """Send message and receive response with timeout and typing indicator handling"""
+        if not self.websocket:
+            await self._connect_websocket()
+        
+        try:
+            # Send message
+            await self.websocket.send(message)
+            logger.debug(f"Sent message: {message[:100]}...")
```

Comment:
This is logging raw content of intentionally malicious prompts that may contain characters that could cause issues or even security impacts for active log watchers.